server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueMailDelete.java

```diff
@@ -59,9 +63,15 @@
     Mono<Void> considerDeleted(MailKey mailKey, MailQueueName mailQueueName) {
         return deletedMailsDao
             .markAsDeleted(mailQueueName, mailKey)
+            .then(decrementBucketSize(mailKey, mailQueueName))
             .doOnNext(ignored -> maybeUpdateBrowseStart(mailQueueName));
     }
 
+    private Mono<Void> decrementBucketSize(MailKey mailKey, MailQueueName mailQueueName) {
+        return mailKeyToBucketDAO.retrieveBucket(mailQueueName, mailKey)
```

Comment:
I'm not too worry about writes but adding a read-before-write to this code path looks scary

Reply:
Agree. 

If we can infer the enqueued date from the mail we can save the additional projection. Maybe adding the slice in the MailReferenceDTO - and carry it along in the RabbitMQXXXItem?

Reply:
Let me answer you on this point:

MailQueueView storage informations is not transmitted over rabbitMQ for good reasons:

 - Publishing to rabbit is done before saving the mail in the view: this means the information is not available 'yet'. Reason consistency: if the rabbit call fails, and we reverse the order the mail will never be cleaned up from the view.

 - We don't wan't to leak Cassandra specific details in the Rabbit specific part of the code anyway - or at the price of costly abstractions.

Now thinking about this "cold-headed', this do not scare me much: it's only very simple metadata. After the queue (where being slow is OK BTW).